File: java/doit_java/Chapter6/src/exercise/Q5/CoffeeTest.java
/*
Q5
아침 출근길에 김 씨는 4,000원을 내고 별다방에서 아메리카노를 사 마셨습니다. 이 씨는 콩다방
에서 4,500원을 내고 라테를 사 마셨습니다. '06-2 객체 간 협력'을 참고하여 이 과정을 객체
지향으로 프로그래밍해 보세요.
 */
package exercise.Q5;

public class CoffeeTest {
    public static void main(String[] args) {

        Person personLim = new Person("임채영", 30000);
        Person personJung = new Person("정흥규", 50000);

        Starbucks starCoffee = new Starbucks();
        Coffeebean beanCoffee = new Coffeebean();

        personLim.buyStarCoffee(starCoffee, 4000);
        personJung.buyBeanCoffee(beanCoffee, 4500);
    }
}

File: python/p_oop2/chap05/excercise/coffee_test.py
from coffee_module.menu import Menu
from coffee_module import *


def main():
    kim = Person("Kim", 10000)
    lee = Person("Lee", 10000)

    star_coffee = StarCoffee()
    bean_coffee = BeanCoffee()

    kim.buy_star_coffee(star_coffee, Menu.STARAMERICANO)
    lee.buy_bean_coffee(bean_coffee, Menu.BEANLATTE)


if __name__ == '__main__':
    main()
